src/Metadata/Driver/XmlDriver.php

```diff
@@ -303,16 +297,18 @@ protected function loadMetadataFromFile(\ReflectionClass $class, string $path):
                         $pMetadata->maxDepth = (int) $pElem->attributes()->{'max-depth'};
                     }
 
+
+                    if (null !== $elem->attributes()->{'read-only'}) {
+                        $metadata->xmlRootPrefix = (string) $xmlRootPrefix;
+                    }
+
                     //we need read-only before setter and getter set, because that method depends on flag being set
-                    $pMetadata->readOnly = $pMetadata->readOnly || $readOnlyClass;
-                    $pMetadata->readOnlyIf = $readOnlyIfClass;
-                    if (null !== $readOnlyIf = $pElem->attributes()->{'read-only-if'}) {
-                        $pMetadata->readOnlyIf = $this->parseExpression((string) $readOnlyIf);
-                        $pMetadata->readOnly = false;
-                    } else {
-                        if (null !== $readOnly = $pElem->attributes()->{'read-only'}) {
-                            $pMetadata->readOnly = 'true' === strtolower((string) $readOnly);
-                            $pMetadata->readOnlyIf = null;
+                    $readOnly = null === $pElem->attributes()->{'read-only'} ? $readOnlyClass : $pElem->attributes()->{'read-only'};
+                    if(null !== $readOnly) {
+                        if ($readOnly === 'true' || $readOnly === 'false') {
```

Comment:
Thought of also parsing the true/false values as expression. But I guess it's not good for performance.

Reply:
here performance are not a concern as this is done only once, then data are cached

Reply:
as I saw it is only caching the expression-string and not the result, which makes sense because the expression should be evaluated on deserialize to be dynamically.
And then it would always evalute expressions with "true" and "false" on deserialize.
Not sure if this will produce impact at all though.